sdks/python/apache_beam/examples/complete/tfidf_test.py

```diff
@@ -85,8 +85,8 @@ def test_basics(self):
     with open_shards(os.path.join(
         temp_folder, 'result-*-of-*')) as result_file:
       for line in result_file:
-        match = re.search(EXPECTED_LINE_RE, line)
-        logging.info('Result line: %s', line)
+        match = re.search(EXPECTED_LINE_RE, line.decode('utf-8'))
```

Comment:
Should open_shards be changed to open as text (decoded) rather than adding decodes everywhere? Or is it used on binary data?

Reply:
I can't find any occurrence where it's used with binary data, so we could do it. But we would probably still want to expose an 'encoding' parameter so we can add tests with different encodings in the future. 

We are also encoding the data in the same file when writing:
https://github.com/apache/beam/blob/11f5efa2bbdf79ac86d321932d6637575904c089/sdks/python/apache_beam/examples/complete/tfidf_test.py#L49-L52
So I like how it's explicit. It is currently the responsibility of the tests to encode/decode correctly.

What do you think?

Reply:
How about we do the following:
1. Make `open_shards` accept glob_pattern + arguments of `io.open()`, using the same defaults.
2. Use `io.open()` instead of builtin `open` in last line of `open_shards` and pass the input arguments.  io.open() interface is consistent between Py2 and Py3.  
3. Make it a responsibility of individual tests whether they want to read text or binary, and/or pass encoding. 

Looks like we won't have to modify test files in this case, since by default we will be opening files as text with default encoding.

Reply:
Sounds good, but I think we need to set `encoding='utf-8'` as a default argument on open_shards, since the default encoding used by `io.open()` is platform dependent.

Reply:
Sounds good, defaulting to `utf-8` will make tests more hermetic.